### PR TITLE
Add string impact options for controls

### PR DESCRIPTION
**Pull request by jquick**
This PR adds the option to specify:
```
none
low
medium
high
critical
```

For your impact section in controls. This will map to their CVSS equivalent number values. This is to help with readability on controls.

Fixes https://github.com/inspec/inspec/issues/562

### CODE REVIEW

**Review by miah — APPROVED** [review]

File: test/unit/impact_test.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    it 'returns true for float string' do
+      impact.is_number?('10.99').must_equal true
+    end
+
+    it 'returns false for char string' do
+      impact.is_number?('abc').must_equal false
+    end
+  end
+
```
Comment (jerryaldrichiii) [inline test/unit/impact_test.rb]: Extra newline, but won't hold up the PR for it.

/me shakes fist at tests not being linted

File: lib/inspec/impact.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  def self.is_number?(value)
+    Float(value)
+    true
+  rescue
+    false
+  end
+
+  def self.string_from_impact(value)
+    value = value.to_f
+    raise Inspec::ImpactError, "'#{value}' is not a valid impact score [0.0 - 1]." if value < 0 || value > 1
```
Comment (jerryaldrichiii) [inline lib/inspec/impact.rb]: Should this be `[0.0 - 1.0]`?

File: lib/inspec/impact.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  }.freeze
+
+  def self.impact_from_string(value)
+    # return if its a number
+    return value if is_number?(value)
+    raise Inspec::ImpactError, "'#{value}' is not a valid impact name [none, low, medium, high, critical]." unless IMPACT_SCORES.key?(value.downcase)
+    IMPACT_SCORES[value]
+  end
+
+  def self.is_number?(value)
```
Comment (jerryaldrichiii) [inline lib/inspec/impact.rb]: You could probably shorten this to:

```
value.is_a?(Integer) || value.is_a?(Float)
```

Comment (jerryaldrichiii) [reply lib/inspec/impact.rb]: Oh...I see...looks like you are attempting to cast as a Float then relying on that not erroring? I'm not following.

File: lib/inspec/impact.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    'low'      => 0.01,
+    'medium'   => 0.4,
+    'high'     => 0.7,
+    'critical' => 0.9,
+  }.freeze
+
+  def self.impact_from_string(value)
+    # return if its a number
+    return value if is_number?(value)
+    raise Inspec::ImpactError, "'#{value}' is not a valid impact name [none, low, medium, high, critical]." unless IMPACT_SCORES.key?(value.downcase)
```
Comment (jerryaldrichiii) [inline lib/inspec/impact.rb]: I'd like to see something like:

```
'#{value}' is not a valid impact name. Options: none, low, medium, high, critical
```

**Review by jerryaldrichiii — COMMENTED** [review]
Couple of suggestions/questions. Won't hold up the PR though.

File: lib/inspec/impact.rb (hunk 3 above)
Comment (jquick) [reply lib/inspec/impact.rb]: Right, if we can cast the string as a Float then its some sort of number and we want to leave it alone.

Comment (jquick) [reply lib/inspec/impact.rb]: Take a look at the test to see how I am using this.

Comment (jerryaldrichiii) [reply lib/inspec/impact.rb]: Hmm...I understand now. Thanks!

**Comment by aaronlippold** [comment]
Hi,

Low is not cast to the correct value is it? 'low' => 0.01, should be 'low'
=> 0.1 right?

Also, what about the values in between? .2, .3, .5, .6 ...

Yours,

Aaron
--------
Aaron Lippold

lippold@gmail.com

260-255-4779

twitter/aim/yahoo,etc.
'aaronlippold'

**Comment by jquick** [comment]
Hey @aaronlippold . Low should be correct at `.01` please take a look at https://www.first.org/cvss/specification-document where it shows the breakdown. values in between will fall into the bracket. So .2/.3 would be low, etc. You can see this here https://github.com/inspec/inspec/blob/c1dfd08d759ac14ac6f0100c765bdc40ec9e2aaa/lib/inspec/impact.rb#L31

**Comment by aaronlippold** [comment]
Hi,

OK I see why you are using .01 i.e. to provide the lower bound... got it.
--------
Aaron Lippold

lippold@gmail.com

260-255-4779

twitter/aim/yahoo,etc.
'aaronlippold'

**Review by jerryaldrichiii — APPROVED** [review]
